pkg/validate/container_service.go

```diff
@@ -38,7 +38,10 @@ func isValidHostname(h string) bool {
 func ContainerService(new, old *api.ContainerService) (errs []error) {
 	// TODO update validation
 	// TODO are these error messages confusing since they may not correspond with the external model?
-	return validateContainerService(new)
+	if errs := validateContainerService(new); len(errs) > 0 {
+		return errs
+	}
+	return validateUpdateContainerService(new, old)
```

Comment:
only if old is not nil

Reply:
I am doing the check inside `validateUpdateContainerService`, no strong feelings on the one over the other.

Reply:
or so I thought

Reply:
maybe my github isn't refreshing, first line of that is an NPE if old is nil

```
func validateUpdateContainerService(cs, oldCs *api.ContainerService) (errs []error) {
	if cs.Location != oldCs.Location {
```